src/Pages/SignUp/SignUp.js

```diff
@@ -0,0 +1,284 @@
+    } else if (name === "re_password") {
+      this.handleRePw(value);
+    }
+  };
+
+  handleEmail = (email) => {
+    const emailRule = /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i;
+    this.setState({
+      emailVal: emailRule.test(email),
+    });
+  };
+
+  handleNumber = (phoneNumber) => {
+    const regExp = /^01(?:0|1|[6-9])(?:\d{3}|\d{4})\d{4}$/;
+    this.setState({
+      numberVal: regExp.test(phoneNumber),
+    });
+  };
+
+  handlePw = (pw) => {
+    console.log(pw.length > 7);
+    pw.length > 7
+      ? this.setState({ passwordVal: true })
+      : this.setState({ passwordVal: false });
+  };
+
+  handleRePw = (re) => {
+    re === this.state.password
+      ? this.setState({ re_passwordVal: true })
+      : this.setState({ re_passwordVal: false });
+  };
+  render() {
+    const {
+      email,
+      emailVal,
+      mobile,
+      numberVal,
+      password,
+      passwordVal,
+      re_passwordVal,
+    } = this.state;
+    return (
+      <div className="SignUp">
+        <nav>
+          <img src="Images/GrafolWEo.png" alt="logo" />
+          <span>
+            <Link to="/Login">로그인</Link>
+          </span>
+        </nav>
+        <div className="signupContainer">
+          <div className="text_box">
+            <p>회원가입</p>
+          </div>
+          <form onSubmit={this.postSignUp}>
+            <div>
+              <label>프로필 이미지</label>
+              <img
+                src="https://previews.123rf.com/images/hydromet/hydromet1211/hydromet121100005/16185946-%ED%95%B4%EB%B3%80%EA%B3%BC-%EB%B0%94%EB%8B%A4.jpg"
+                alt="profile_img"
+              />
+              <div className="idBox">
+                <input
+                  autoComplete="off"
+                  placeholder="url주소를 입력해주세요."
+                />
+              </div>
+            </div>
+            <div className="idBox">
+              <label>이름</label>
+              <input
+                autoComplete="off"
+                onKeyUp={this.checkVal}
+                name="user_name"
+                type="text"
+                placeholder="홍길동"
+              />
+            </div>
+            <div className="idBox">
+              <label>이메일</label>
+              <input
+                autoComplete="off"
+                onKeyUp={this.checkVal}
+                name="email"
+                type="email"
+                placeholder="example@naver.com"
+              />
+            </div>
+            {email.length !== 0 && !emailVal && (
+              <div className="alertMessage">
+                <span role="img" aria-label="">
+                  🔺 올바른 이메일이 아닙니다.
+                </span>{" "}
+              </div>
+            )}
+            {email.length === 0 && !emailVal && (
+              <div className="alertMessage">
+                <span role="img" aria-label="">
+                  🔺 이메일을 입력해 주세요.
+                </span>{" "}
+              </div>
+            )}
```

Comment:
'email.length === 0 이라는 코드는 email.length가 0일때를 의미합니다. 숫자 0은 boolean값이 false입니다. 그래서 아래와 같이 코드를 작성해주실 수 있습니다. :)

```suggestion
            {!email.length && !emailVal && (
              <div className="alertMessage">
                <span role="img" aria-label="">
                  🔺 이메일을 입력해 주세요.
                </span>{" "}
              </div>
            )}
```

Reply:
우와 감사합니다.ㅎㅎ